src/pages/connect/choose-account.tsx

```diff
@@ -31,11 +31,21 @@ export const ChooseAccount: React.FC<ChooseAccountProps> = ({ next }) => {
   const [reusedApps, setReusedApps] = React.useState<ConfigApp[]>([]);
   const { decodedAuthRequest: authRequest } = useOnboardingState();
   const [accountIndex, setAccountIndex] = React.useState<number | undefined>();
+  const { handleCancelAuthentication } = useWallet();
 
   if (!wallet) {
     return <Navigate to={{ pathname: '/', hash: 'sign-up' }} screenPath={ScreenPaths.GENERATION} />;
   }
 
+  const handleUnmount = useCallback(async () => {
+    handleCancelAuthentication();
+  }, [handleCancelAuthentication]);
+
+  useEffect(() => {
+    window.addEventListener('beforeunload', handleUnmount);
+    return () => window.removeEventListener('beforeunload', handleUnmount);
+  }, [handleUnmount]);
+
```

Comment:
This is the only component I found during auth that doesn't use the `PopupContainer` so not great to have duplicate code here. Can this be wrapped in the `PopupContainer`? Why is this a `Screen`?